### PR TITLE
fix: various cleanup and logs

### PR DESCRIPTION
**Pull request by stringhandler**

### CODE REVIEW

File: src/server/config.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ pub struct Config {
 impl Default for Config {
     fn default() -> Self {
         Self {
-            base_node_address: String::from("http://127.0.0.1:18142"),
+            base_node_address: String::from("http://127.0.0.1:18182"),
```
Comment (SWvheerden) [inline src/server/config.rs:30]: we should get this from the network, each network has a different one

File: src/sharechain/in_memory.rs [hunk 2]
```diff
@@ -115,29 +115,30 @@ impl InMemoryShareChain {
...
+                        &block.original_block_header,
                         params.random_x_factory(),
                         params.genesis_block_hash(),
                         params.consensus_manager(),
                     )
                     .map_err(Error::RandomXDifficulty)?;
                     Ok(difficulty.as_u64())
                 } else {
-                    Ok(0)
+                    panic!("No params provided for RandomX difficulty calculation!");
```
Comment (hansieodendaal) [inline src/sharechain/in_memory.rs:136]: Why not return an error here?

File: src/sharechain/mod.rs [hunk 3]
```diff
@@ -16,7 +16,7 @@ use crate::sharechain::{block::Block, error::Error};
 pub const CHAIN_ID: usize = 1;
 
 /// How many blocks to keep overall.
-pub const MAX_BLOCKS_COUNT: usize = 600;
+pub const MAX_BLOCKS_COUNT: usize = usize::MAX;
```
Comment (SWvheerden) [inline src/sharechain/mod.rs:19]: we should use an lwma, and store 2* the max lwma lenght

**Review by SWvheerden — APPROVED** [review]

File: src/sharechain/mod.rs (hunk 3 above)
Comment (hansieodendaal) [inline src/sharechain/mod.rs:19]: Why not just 800 or 1200?

**Review by hansieodendaal — COMMENTED** [review]
Some comments

**Pull request merged**